File: billing-domain/src/main/java/com/example/billing/domain/webhook/WebhookEndpointId.java
package com.example.billing.domain.webhook;

import java.util.Objects;
import java.util.UUID;

public record WebhookEndpointId(UUID value) {
    public WebhookEndpointId { Objects.requireNonNull(value, "WebhookEndpointId.value"); }
    public static WebhookEndpointId newId() { return new WebhookEndpointId(UUID.randomUUID()); }
    public static WebhookEndpointId of(String s) { return new WebhookEndpointId(UUID.fromString(s)); }
    @Override public String toString() { return value.toString(); }
}

File: billing-domain/src/main/kotlin/com/example/billing/domain/webhook/WebhookEndpointId.kt
package com.example.billing.domain.webhook

import java.util.UUID

/**
 * Webhook endpoint 식별자.
 *
 * Java 호환을 위해 value class 가 아닌 일반 data class. `@get:JvmName("value")` 로 기존
 * Java record accessor `value()` 보존.
 */
data class WebhookEndpointId(@get:JvmName("value") val value: UUID) {

    override fun toString(): String = value.toString()

    companion object {
        @JvmStatic
        fun newId(): WebhookEndpointId = WebhookEndpointId(UUID.randomUUID())

        @JvmStatic
        fun of(s: String): WebhookEndpointId = WebhookEndpointId(UUID.fromString(s))
    }
}
